pgmpy/readwrite/BIF.py

```diff
@@ -0,0 +1,312 @@
+import numpy as np
+from pyparsing import Word, alphanums, Suppress, Optional, CharsNotIn, Group, nums, ZeroOrMore, OneOrMore, cppStyleComment, Literal
+from pgmpy.models import BayesianModel
+from pgmpy.factors import TabularCPD
+import re
+
+
+class BifReader(object):
+
+    """
+    Base class for reading network file in bif format
+    """
+    def __init__(self, path=None, string=None):
+
+        """
+        Initialisation of BifReader object
+
+        Parameters
+        ----------------
+        path : file or str
+                File of bif data
+        string : str
+                String of bif data
+        Examples
+        -----------------
+        # dog-problem.bif file is present at
+        # http://www.cs.cmu.edu/~javabayes/Examples/DogProblem/dog-problem.bif
+        >>> reader = BifReader("bif_test.bif")
```

Comment:
@khalibartan Should be a working example. So add import statement and maybe fetch the file and then read it.